From: ServBay Team
Subject: Fix libxml2 2.12+ compatibility for PHP 8.0

The libxml2 2.12+ changed xmlStructuredErrorFunc to use const xmlError*.
This patch fixes the compatibility issue.

---
 ext/libxml/libxml.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/ext/libxml/libxml.c b/ext/libxml/libxml.c
--- a/ext/libxml/libxml.c
+++ b/ext/libxml/libxml.c
@@ -732,7 +732,11 @@ PHP_LIBXML_API void php_libxml_ctx_warning(void *ctx, const char *msg, ...)
 	va_end(args);
 }

-PHP_LIBXML_API void php_libxml_structured_error_handler(void *userData, xmlErrorPtr error)
+#if LIBXML_VERSION >= 21200
+PHP_LIBXML_API void php_libxml_structured_error_handler(void *userData, const xmlError *error)
+#else
+PHP_LIBXML_API void php_libxml_structured_error_handler(void *userData, xmlErrorPtr error)
+#endif
 {
 	_php_list_set_error_structure(error, NULL);

@@ -1034,7 +1038,11 @@ PHP_FUNCTION(libxml_use_internal_errors)
 /* {{{ Retrieve last error from libxml */
 PHP_FUNCTION(libxml_get_last_error)
 {
-	xmlErrorPtr error;
+#if LIBXML_VERSION >= 21200
+	const xmlError *error;
+#else
+	xmlErrorPtr error;
+#endif

 	ZEND_PARSE_PARAMETERS_NONE();

